src/main/kotlin/net/maizegenetics/phgv2/api/HaplotypeGraph.kt

```diff
@@ -209,15 +220,14 @@ class HaplotypeGraph(hvcfFiles: List<String>) {
      */
     private suspend fun addSites() {
 
-        // rangeIdToSampleToChecksum[refRangeId][sampleId] -> Checksum / hapid
-        val rangeIdToSampleToChecksum = mutableListOf<Array<String?>>()
+        // rangeIdToSampleToChecksum[refRangeId][sampleId][gameteId] -> Checksum / hapid
+        val rangeIdToSampleToChecksum = mutableListOf<Array<MutableList<String?>>>()
```

Comment:
Is there a way to get away from nullable Strings?  Can we use empty strings instead?

Reply:
Ok, I changed HaplotypeGraph.rangeByGameteIdToHapid to use "" instead of nulls. Some intermediate variables are still using null but not exposed to the user of the class.